client-apis/metrics-api/pom.xml

```diff
@@ -61,6 +61,10 @@
       <groupId>org.wildfly.swarm</groupId>
       <artifactId>cdi</artifactId>
     </dependency>
+    <dependency>
```

Comment:
Does this mean the client-api is depending on the Fraction?

I think that's backwards from what we've done in the past

Reply:
Yes, metrics-api (which is actually not a client API but merely a CDI extension) depends on fraction API, namely on https://github.com/mkouba/wildfly-swarm/blob/SWARM-1641/fractions/microprofile/microprofile-metrics/src/main/java/org/wildfly/swarm/microprofile/metrics/api/RegistryFactory.java. The reason is that the CDI extension does a JNDI lookup to find a registry factory provided by fraction. We can improve this later...

Reply:
So should the Metrics Client API actually live in https://github.com/wildfly-swarm/wildfly-swarm/tree/master/fractions/cdi-extensions?

I ask because what's present in client-api are not things that we expect a user to directly add a dependency for, but this sounds like it's something that should be? Am I wrong?

Reply:
Yes, I think it should live somewhere else. For me "client API" is something a user would like to communicate with and that's what `metrics-api` is not. See also https://issues.jboss.org/browse/SWARM-1631?focusedCommentId=13496588&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13496588

`metrics-api` should be probably called `metrics-cdi-extension`. But it's not a fraction so I'm not sure whether it should live in `fractions/cdi-extensions`.